Task_4/Task_4.1/LogService.cs

```diff
@@ -0,0 +1,236 @@
+
+        private const string _pathDirectoryStates = "C:\\LogState";
+
+        /// <summary>
+        /// Current folder for researching
+        /// </summary>
+        public string CurrentPath { get; private set; }
+
+        /// <summary>
+        /// Path folder for keeping content of files
+        /// </summary>
+        public string LogContentPathFolder => CurrentPath + _nameFolderLogContent;
+
+        /// <summary>
+        /// Path folder for keeping commite of fixation
+        /// </summary>
+        public string FixationLogPathFolder => CurrentPath + _nameFolderFixationLog;
+
+        public LogService(string currentFolderPath)
+        {
+            CurrentPath = currentFolderPath;
+        }
+
+        public static List<Log> GetListLog(string pathLog)
+        {
+            return Deserialize(pathLog);
+        }
+
+        /// <summary>
+        /// Return content selected file by commite
+        /// </summary>
+        /// <param name="id">Id commite</param>
+        /// <returns></returns>
+        public string GetContentLogById(Guid id)
+        {
+            string pathContent = $"{LogContentPathFolder}\\{id}.json";
+
+            return !File.Exists(pathContent) ? File.ReadAllText(pathContent) : null;
+        }
+
+        /// <summary>
+        /// Save list commites by current status
+        /// </summary>
+        /// <param name="list">list commites</param>
+        /// <param name="pathLog">path current fixation</param>
+        public void SetListLog(List<Log> list, string pathLog)
+        {
+            if (list.Count > 0)
+            {
+                Serialize(list, pathLog);
+
+                if (!Directory.Exists(LogContentPathFolder))
+                {
+                    Directory.CreateDirectory(LogContentPathFolder);
+                }
+
+                //save content files
+                foreach (var log in list)
+                {                    
+                    if (log.Type != LogType.Delete || log.Type != LogType.Rename)
+                    {
+                        string pathLogContent = $"{LogContentPathFolder}\\{log.Id}.json";
+
+                        File.WriteAllText(pathLogContent, log.Content);
+                    }
+                }
+            }
+        }
+
+        /// <summary>
+        /// Return all have fixations
+        /// </summary>
+        /// <returns></returns>
+        public List<Log> GetAllFixation()
+        {
+            if (Directory.Exists(FixationLogPathFolder))
+            {
+                var files = Directory.GetFiles(FixationLogPathFolder, "*.json");
+                if (files.Count() > 0)
+                {
+                    var filesBefore = new Dictionary<string, DateTime>();
+                    foreach (var item in files)
+                    {
+                        filesBefore.Add(item, File.GetLastWriteTime(item));
+                    }
+                    var filesAfter = filesBefore.OrderBy(x => x.Value).ToList();
```

Comment:
* what is `before` and `after`?

```suggestion
                    var filesByDate = files
                        .Select(path => new {path, WriteTime = File.GetLastWriteTime(item)})
                        .OrderBy(x => x.writeTime)
                        .ToList();
                    
```